internal/resources/metal/vrf_bgp_dynamic_neighbor/resource.go

```diff
@@ -0,0 +1,174 @@
+				Name: "equinix_metal_vrf_bgp_dynamic_neighbor",
+			},
+		),
+	}
+
+	return &r
+}
+
+func (r *Resource) Schema(
+	ctx context.Context,
+	req resource.SchemaRequest,
+	resp *resource.SchemaResponse,
+) {
+	s := resourceSchema(ctx)
+	if s.Blocks == nil {
+		s.Blocks = make(map[string]schema.Block)
+	}
+
+	resp.Schema = s
+}
+
+func (r *Resource) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
+	client := r.Meta.NewMetalClientForFramework(ctx, request.ProviderMeta)
+
+	var plan Model
+	response.Diagnostics.Append(request.Config.Get(ctx, &plan)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	createRequest := metalv1.BgpDynamicNeighborCreateInput{
+		BgpNeighborRange: plan.Range.ValueString(),
+		BgpNeighborAsn:   plan.ASN.ValueInt64(),
+	}
+
+	response.Diagnostics.Append(getPlanTags(ctx, plan, &createRequest.Tags)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	neighbor, _, err := client.VRFsApi.CreateBgpDynamicNeighbor(ctx, plan.GatewayID.ValueString()).
+		BgpDynamicNeighborCreateInput(createRequest).
+		Exclude(bgpNeighborExcludes).
+		Include(bgpNeighborIncludes).
+		Execute()
+
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Error creating VRF BGP dynamic neighbor range",
+			"Could not create VRF BGP dynamic neighbor range: "+err.Error(),
+		)
+	}
+
+	// Parse API response into the Terraform state
+	response.Diagnostics.Append(plan.parse(ctx, neighbor)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	// Set state to fully populated data
+	response.Diagnostics.Append(response.State.Set(ctx, &plan)...)
+}
+
+func (r *Resource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	client := r.Meta.NewMetalClientForFramework(ctx, request.ProviderMeta)
+
+	var data Model
+	response.Diagnostics.Append(request.State.Get(ctx, &data)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	neighbor, _, err := client.VRFsApi.BgpDynamicNeighborsIdGet(ctx, data.ID.ValueString()).
+		// `created_by` is specified as a UserLimited.  To avoid an error
+		// due to missing UserLimited.id field, have to either exclude
+		// or include `created_by`
+		Exclude(bgpNeighborExcludes).
+		Include(bgpNeighborIncludes).
+		Execute()
+
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Error reading VRF BGP dynamic neighbor range",
+			"Could not read VRF BGP dynamic neighbor with ID "+data.ID.ValueString()+": "+err.Error(),
+		)
+	}
+
+	response.Diagnostics.Append(data.parse(ctx, neighbor)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	response.Diagnostics.Append(response.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// TODO: it should be possible to update tags but the API spec doesn't
+	// mention support for that.  In the meantime should changes to tags force
+	// recreating the resource?
```

Comment:
Is there an update endpoint for VRF BGP dynamic neighbors?  I don't see one in the spec, but since the resource has tags it seems like it should be possible to update those, at least.

Reply:
If there _is_ an update endpoint and it allows changes that modify the `status` of the range, do we need to wait for a particular `status` here?

Reply:
Confirmed that there is no update endpoint in the API.  Comment has been updated to reflect that, and `tags` has been configured to force a new resource on changes.